cargo-auditable/build.rs

```diff
@@ -0,0 +1,107 @@
+
+Yes. The data format is designed for interoperability with alternative implementations. In fact, parsing it only takes 5 lines of Python. See https://github.com/rust-secure-code/cargo-auditable/blob/HEAD/PARSING.md for documentation on parsing the data.
+")
+            .paragraph("What is the data format, exactly?
+
+The data format is described by the JSON schema https://github.com/rust-secure-code/cargo-auditable/blob/HEAD/cargo-auditable.schema.json. The JSON is Zlib-compressed and placed in a linker section named .dep-v0. You can find more info about parsing it here.
+")
+            .paragraph("What about embedded platforms?
+
+Embedded platforms where you cannot spare a byte should not add anything in the executable. Instead they should record the hash of every executable in a database and associate the hash with its Cargo.lock, compiler and LLVM version, build date, etc. This would make for an excellent Cargo wrapper or plugin. Since that can be done in a 5-line shell script, writing that tool is left as an exercise to the reader.
+")
+            .paragraph("Does this impact reproducible builds?
+
+The data format is specifically designed not to disrupt reproducible builds. It contains no timestamps, and the generated JSON is sorted to make sure it is identical between compilations. If anything, this helps with reproducible builds, since you know all the versions for a given binary now.
+")
+            .paragraph("Does this disclose any sensitive information?
+
+No. All URLs and file paths are redacted, but the crate names and versions are recorded as-is. At present panic messages already disclose all this info and more. Also, chances are that you're legally obligated have to disclose use of specific open-source crates anyway, since MIT and many other licenses require it.
+")
+            .paragraph("What about recording the compiler version?
+
+The compiler itself will start embedding it soon.
+
+On older versions it's already there in the debug info. On Unix you can run strings your_executable | grep 'rustc version' to see it.
+")
+            .paragraph("What about keeping track of versions of statically linked C libraries?
+
+Good question. I don't think they are exposed in any reasonable way right now. Would be a great addition, but not required for the initial launch. We can add it later in a backwards-compatible way. Adopting the -src crate convention would make it happen naturally, and will have other benefits as well, so that's probably the best route.
+")
+            .paragraph("What is blocking uplifting this into Cargo?
+
+Cargo itself is currently in a feature freeze.
+")
+        )
+        .render()
+}
+
+fn generate_man_page_file() -> Result<()> {
+    let mut dest_path = std::env::current_exe()?;
+    dest_path.pop();
+    dest_path.pop();
+    dest_path.pop();
+    dest_path.pop();
+    dest_path.push("man-page");
+    dest_path.push("cargo-auditable");
```

Comment:
This will likely cause issues in some configurations. Shouldn't you simply read the `OUT_DIR` environment variable that [Cargo sets for builds scripts?](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts)

Reply:
That seems better, thanks :)